dali/operators/image/resize/resampling_attr.cc

```diff
@@ -0,0 +1,122 @@
+// Copyright (c) 2019-2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "dali/operators/image/resize/resampling_attr.h"
+#include "dali/pipeline/operator/common.h"
+
+namespace dali {
+
+DALI_SCHEMA(ResamplingFilterAttr)
+  .DocStr(R"code(Resampling filter attribute placeholder)code")
+  .AddOptionalArg("interp_type",
+      R"code(Type of interpolation used. Use `min_filter` and `mag_filter` to specify
+different filtering for downscaling and upscaling.)code",
+      DALI_INTERP_LINEAR, true)
+  .AddOptionalArg("mag_filter", "Filter used when scaling up",
+      DALI_INTERP_LINEAR, true)
+  .AddOptionalArg("min_filter", "Filter used when scaling down",
+      DALI_INTERP_LINEAR, true)
+  .AddOptionalArg<DALIDataType>("dtype", "Output data type; must be same as input type or `float`. "
+     "If not set, input type is used.", nullptr)
+  .AddOptionalArg("temp_buffer_hint",
+      "Initial size, in bytes, of a temporary buffer for resampling.\n"
+      "Ingored for CPU variant.\n",
+      0)
+  .AddOptionalArg("minibatch_size", "Maximum number of images processed in a single kernel call",
+      32);
+
+
+using namespace kernels;  // NOLINT
+
+inline ResamplingFilterType interp2resample(DALIInterpType interp) {
+#define DALI_MAP_INTERP_TO_RESAMPLE(interp, resample) case DALI_INTERP_##interp:\
+  return ResamplingFilterType::resample;
+
+  switch (interp) {
+    DALI_MAP_INTERP_TO_RESAMPLE(NN, Nearest);
+    DALI_MAP_INTERP_TO_RESAMPLE(LINEAR, Linear);
+    DALI_MAP_INTERP_TO_RESAMPLE(CUBIC, Cubic);
+    DALI_MAP_INTERP_TO_RESAMPLE(LANCZOS3, Lanczos3);
+    DALI_MAP_INTERP_TO_RESAMPLE(GAUSSIAN, Gaussian);
+    DALI_MAP_INTERP_TO_RESAMPLE(TRIANGULAR, Triangular);
+  default:
+    DALI_FAIL("Unknown interpolation type");
+  }
+#undef DALI_MAP_INTERP_TO_RESAMPLE
+}
+
+void ResamplingFilterAttr::PrepareFilterParams(
+      const OpSpec &spec, const ArgumentWorkspace &ws, int num_samples) {
+  GetPerSampleArgument(interp_type_arg_, "interp_type", spec, ws, num_samples);
+  GetPerSampleArgument(min_arg_, "min_filter", spec, ws, num_samples);
+  GetPerSampleArgument(mag_arg_, "mag_filter", spec, ws, num_samples);
+  if (!spec.TryGetArgument(dtype_arg_, "dtype"))
```

Comment:
If you set the default to DALI_NO_TYPE as I suggested above you can just do `spec.GetArgument`